src/understand-tidb/cbo.md

```diff
@@ -1 +1,165 @@
+
+In the figure, the black font operator is a logical operator(e.g, `Agg`, `Join` and `DS`), the blue font is a physical operator(e.g, `Stream Agg`, `Hash Join` and `Index Scan(a)`...), and the yellow arrow is an operator for which the cost has been calculated. And the red dashed arrow is not in compliance with `prop` operator. The font on the arrow describes the property requirements for the lower operators(e.g, `s.a` means the the output of the lower operator needs to be ordered according to `s.a`).
+
+1. `Step1`: The `getTask` function corresponds to the yellow arrow in the figure, which means that the calculated part can be used directly without repeated calculations.
+2. `Step2`: The `p.self.exhaustPhysicalPlans(prop)` function in the figure represents the process of logical operator generating physical operator. For example, logical operator `Agg` generates physical operators `Stream Agg` and `Hash Agg`. But some physical operators that do not satisfy the required property cannot be generated. For example, if the required property is `s.a` logical operator `DS`, only physical operator `Index Scan(a)` can be generated, but `TableScan` cannot be generated.
+3. `Step3`: After we generate all the possible physical plans, we should compare their cost and choose the lowest cost plan. For example, for the plan of `Stream Agg -> Sort Merge Join -> Index Scan(a)`, its cost is `Cost(Stream Agg) + Cost(Sort Merge Join) + Cost(Index Scan(a))`.
+
+![find-best-plan](../img/cbo-explain-getBestTask.png)
+
+#### Exhaust Physical Plans
+
+For each operator, the logic for exhausting physical plans are in `exhaustPhysicalPlans(*property.PhysicalProperty) (physicalPlans []PhysicalPlan)` method of the `LogicalPlan` interface. Their specific implementation is in the [`planner/core/exhaust_physical_plans.go`](https://github.com/pingcap/tidb/blob/master/planner/core/exhaust_physical_plans.go) file. 
+
+`exhaustPhysicalPlans` generates all possible plans that can match the required property. Different operators have different implementations. You can learn more about how different logical operators generate corresponding physical operators based on the required property by reading the code.
+
+#### Enumerate Physical Plans For Task
+
+`Enumerate Physical Plans For Task` will use the dynamic programming algorithm to select the lowest cost plan among all generated physical plans and it will return the `task` which encapsulates the selected physical plan. The `task` is a new version of physical plan. It stores cost information for a physical plan. You can see the interface definition of task in the [`planner/core/task.go`](https://github.com/pingcap/tidb/blob/master/planner/core/task.go)  file. A task may be `CopTask`,  `RootTask`, or a `MPPTask`. `copTask` is a task that runs in a distributed kv store. `rootTask` is the task that runs in the `TiDB` compute layer. `MPPTask` is a task that related to the mpp plan.
```

Comment:
```suggestion
`Enumerate Physical Plans For Task` will use the dynamic programming algorithm to select the lowest cost plan among all generated physical plans and it will return the `task` which encapsulates the selected physical plan. The `task` is a new version of physical plan. It stores cost information for a physical plan. You can see the interface definition of task in the [`planner/core/task.go`](https://github.com/pingcap/tidb/blob/master/planner/core/task.go) file. A task may be `copTask`,  `rootTask`, or a `mppTask`. `copTask` is a task that runs in a distributed kv store. `rootTask` is the task that runs in the `TiDB` compute layer. `mppTask` is a task that related to the mpp plan.
```